minilib/src/main/scala/java/lang/System.scala

```diff
@@ -0,0 +1,17 @@
+package java.lang
+
+object System {
+  // Simpler than the original, technically valid but of lesser quality
+  def identityHashCode(x: Object): scala.Int = {
+    (x: Any) match {
+      case null => 0
+      case _:scala.Boolean | _:scala.Double | _:String | () =>
+        x.hashCode()
+      case _ =>
+        if (x.getClass == null)
```

Comment:
isn't this condition inverted?

Reply:
No, it isn't. See https://github.com/sjrd/scala-js/blob/a52281c2acce11e6a40f2052d9d64a76f34bc5df/javalanglib/src/main/scala/java/lang/System.scala#L163